### PR TITLE
Fix inconsistency in keeping track of NuGetVersion.OriginalVersion

### PR DESCRIPTION
**Pull request by xavierdecoster**
Fixes NuGet/Home#5068

### CODE REVIEW

File: src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs
Comment (xavierdecoster) [inline src/NuGet.Core/NuGet.Versioning/NuGetVersion.cs:31]: This was the culprit.

File: test/NuGet.Core.Tests/NuGet.Versioning.Test/NuGetVersionTest.cs
Comment (xavierdecoster) [inline test/NuGet.Core.Tests/NuGet.Versioning.Test/NuGetVersionTest.cs:295]: Without the above fix, these three SemVer2 versions would fail the test.

**Review by joelverhagen — APPROVED** [review]
Please wait for a client person's sign-off (@emgarten).

**Review by emgarten — APPROVED** [review]
LGTM

**Pull request merged**
